richtextfx-demos/src/main/java/org/fxmisc/richtext/demo/hyperlink/TextHyperlinkArea.java

```diff
@@ -0,0 +1,69 @@
+package org.fxmisc.richtext.demo.hyperlink;
+
+import javafx.geometry.VPos;
+import org.fxmisc.richtext.GenericStyledArea;
+import org.fxmisc.richtext.TextExt;
+import org.fxmisc.richtext.model.ReadOnlyStyledDocument;
+import org.fxmisc.richtext.model.StyledText;
+import org.fxmisc.richtext.model.TextOps;
+import org.reactfx.util.Either;
+
+import java.util.function.Consumer;
+
+public class TextHyperlinkArea extends GenericStyledArea<ParStyle, Either<StyledText<TextStyle>, Hyperlink<TextStyle>>, TextStyle> {
```

Comment:
> Not to disparage your work, it's great that you put this together.

None taken. I appreciate your feedback!

You are correct. `ParStyle` isn't needed at all.

If Java supported algebraic data types, writing the `SegmentOps` and real/empty hyperlink data types would be a lot easier to code... but it doesn't :-/